app/models/spree/variant.rb

```diff
@@ -243,5 +245,10 @@ def ensure_unit_value
 
       self.unit_value = 1.0
     end
+
+    def convert_variant_weight_to_decimal
+      self.weight = weight.to_d if weight.present?
+      self.weight = 0.0 if weight.nil?
```

Comment:
I think we can just do:
```suggestion
      self.weight = weight.to_d
```
and if it's not present (nil) it'll still be converted to 0.0 anyway.

Reply:
Oh! less code. Neat!